salt/dex/roles.yaml

```diff
@@ -0,0 +1,47 @@
+---
+# This role exists so that the CLI can find Dex at
+# any time. It will be in a different location in a
+# cloud provider environment.
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: find-dex
+  namespace: kube-system
+rules:
+- apiGroups: [""]
+  resources: ["services"]
+  resourceNames: ["dex"]
+  verbs: ["get"]
+---
+# Allow any authenticated *or* unauthenticated
+# user to look up Dex's service entry
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: find-dex
+  namespace: kube-system
+subjects:
+- kind: Group
+  name: system:authenticated
+  apiGroup: rbac.authorization.k8s.io
+- kind: Group
+  name: system:unauthenticated
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: Role
+  name: find-dex
+  apiGroup: rbac.authorization.k8s.io
+---
+# Map the LDAP Administrators role to the Kubernetes system:masters group
```

Comment:
This feels like it shouldn't be part of the base install, as not all LDAP's will have an "Administrators" group which is intended to map to the system:masters role.

Maybe making this a pillar value is enough?

Reply:
(This probably falls into the same "next PR" as the pre-existing LDAP comment below...)